portality/bll/services/application.py

```diff
@@ -561,3 +570,209 @@ def delete_application(self, application_id, account):
+                journal_questions = Journal2PublisherUploadQuestionsXwalk.journal2question(j)
+
+                if len(updates) == 0:
+                    validation.row(validation.WARN, row_ix, "The data you supplied did not change any of the existing values in the journal record.  You may still submit this record, but no update will be made to the record in DOAJ.")
+                    continue
+
+                # if we get to here, then there are updates
+
+                [validation.log(upd) for upd in updates]
+
+                # If a field is disabled in the UR Form Context, then we must confirm that the form data from the
+                # file has not changed from that provided in the source
+                formulaic_context = ApplicationFormFactory.context("update_request")
+                disabled_fields = formulaic_context.disabled_fields()
+                trip_wire = False
+                for field in disabled_fields:
+                    field_name = field.get("name")
+                    question = Journal2PublisherUploadQuestionsXwalk.q(field_name)
+                    update_value = update_form.get(field_name)
+                    journal_value = journal_form.get(field_name)
+                    if update_value != journal_value:
+                        trip_wire = True
+                        validation.value(validation.ERROR, row_ix, header_row.index(question),
+                                         f'"{question}" cannot be changed.',
+                                         was=journal_value, now=update_value)
+
+                if trip_wire:
+                    continue
+
+                # Create an update request for this journal
+                update_req = None
+                jlock = None
+                alock = None
+                try:
+                    # ~~ ^->UpdateRequest:Feature ~~
+                    update_req, jlock, alock = self.update_request_for_journal(j.id, account=j.owner_account, lock_records=False)
+                except AuthoriseException as e:
+                    validation.row(validation.ERROR, row_ix, 'Could not create update request: {0}'.format(e.reason))
+                    continue
+
+                # If we don't have a UR, we can't continue
+                if update_req is None:
+                    validation.row(validation.ERROR, row_ix,'Journal not in DOAJ - missing or not public')
+                    continue
+
+                # validate update_form - portality.forms.application_processors.PublisherUpdateRequest
+                # ~~ ^->UpdateRequest:FormContext ~~
+                fc = formulaic_context.processor(
+                    formdata=update_form,
+                    source=update_req
+                )
+
+                if not fc.validate():
+                    for k, v in fc.form.errors.items():
+                        question = Journal2PublisherUploadQuestionsXwalk.q(k)
+                        pos = header_row.index(question)
```

Comment:
Can raise `ValueError` when header row is outdated - throws a 500 - need a try/catch

```
  File "/home/steve/code/cl/doaj/portality/view/publisher.py", line 400, in journal_csv_validate
    report = DOAJ.applicationService().validate_update_csv(out, current_user)
  File "/home/steve/code/cl/doaj/portality/bll/services/application.py", line 695, in validate_update_csv
    pos = header_row.index(question)
ValueError: 'URL for deposit policy' is not in list
```

Discovered when throwing an old Wiley JournalCSV update at it.